frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -643,6 +715,34 @@ export const transformKnNodeData = (
         const data = createTopologyNodeData(item, type, getImageForIconClass(`icon-openshift`));
         knDataModel.nodes.push(...getKnativeTopologyNodeItems(res, type, data, resources));
         knDataModel.edges.push(...getEventTopologyEdgeItems(res, resources.ksservices));
+        // form node data for sink uri
+        const sinkUri = res.spec?.sink?.uri;
+        let sinkTargetUid = getSinkTargetUid(knDataModel.nodes, sinkUri);
+        if (sinkUri) {
+          if (!sinkTargetUid) {
+            sinkTargetUid = `${res?.metadata?.uid}_nodesinkuri`;
+            const eventSourcesData = getEventSourcesData(sinkUri, resources);
+            const sinkUriObj = {
+              metadata: {
+                uid: sinkTargetUid,
+                namespace: data.resources.obj.metadata.namespace || '',
+              },
+              spec: { sinkUri },
+              type: { nodeType: NodeType.SinkUri },
+            };
+            const sinkData = {
+              id: sinkTargetUid,
+              name: 'URI',
+              type: NodeType.SinkUri,
+              resources: { ...data.resources, obj: sinkUriObj, eventSources: eventSourcesData },
+              data: { ...data.data, sinkUri },
+            };
+            knDataModel.nodes.push(
+              ...getSinkUriTopologyNodeItems(res, NodeType.SinkUri, sinkTargetUid, sinkData),
```

Comment:
The sink URI ID should be based on the URI itself so that if multiple nodes are connected to the same URI.
The side effect of not doing so can be seen when you move a sink connector from another node to an existing URI node. The URI node gets re-laid out.